Mage/src/main/java/mage/abilities/effects/common/continuous/PlayerCanOnlyAttackInDirectionReplacementEffect.java

```diff
@@ -0,0 +1,104 @@
+package mage.abilities.effects.common.continuous;
+
+import mage.abilities.Ability;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.ReplacementEffectImpl;
+import mage.abilities.effects.common.ChooseModeEffect;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.players.PlayerList;
+
+/**
+ * @author TheElk801, Susucr
+ */
+public class PlayerCanOnlyAttackInDirectionReplacementEffect extends ReplacementEffectImpl {
+
+    static final String ALLOW_ATTACKING_LEFT = "Allow attacking left";
+    static final String ALLOW_ATTACKING_RIGHT = "Allow attacking right";
+
+    public PlayerCanOnlyAttackInDirectionReplacementEffect(Duration duration, String directionText) {
+        super(duration, Outcome.Neutral);
+        staticText = duration + (duration.toString().isEmpty() ? "" : ", ")
+                + "each player may attack only the nearest opponent "
+                + "in " + directionText + " and planeswalkers controlled by that opponent";
+    }
+
+    private PlayerCanOnlyAttackInDirectionReplacementEffect(PlayerCanOnlyAttackInDirectionReplacementEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public PlayerCanOnlyAttackInDirectionReplacementEffect copy() {
+        return new PlayerCanOnlyAttackInDirectionReplacementEffect(this);
+    }
+
+    public static Effect choiceEffect() {
+        return new ChooseModeEffect(
+                "Choose a direction to allow attacking in.",
+                ALLOW_ATTACKING_LEFT, ALLOW_ATTACKING_RIGHT
+        ).setText("choose left or right");
+    }
+
+    @Override
+    public boolean replaceEvent(GameEvent event, Ability source, Game game) {
+        return true;
```

Comment:
Well this is the whole root of the problem right? Why is it a replacement effect rather than a restriction?

Reply:
I need to read on Restriction more. But definitively that makes a lot of sense.